bookwyrm/views/reading.py

```diff
@@ -141,16 +149,18 @@ def update_readthrough(request, book=None, create=True):
     start_date = request.POST.get("start_date")
     if start_date:
         try:
-            start_date = timezone.make_aware(dateutil.parser.parse(start_date))
-            readthrough.start_date = start_date
+            readthrough.start_date = load_date_in_user_tz_as_utc(
```

Comment:
I wonder if this is the best place to intervene with the timezone? If I'm understanding this right, the user's preferred timezone is treated as utc, and stored accordingly in the database. As a result, when the template to select read dates is rendered, "now" still appears as a UTC date, and on save, the date that gets stored would be inaccurate because of the translation between timezones.

If the timezone gets configured in the template, I think the user can select the date in their preferred timezone, and it will be correctly annotated to save (https://docs.djangoproject.com/en/3.1/topics/i18n/timezones/#selecting-the-current-time-zone). 

That said, I'm fuzzy on how all this works and might be completely off base!

Reply:
What I have learned in my experience is that you want to be super consistent in how you persist. UTC everywhere in the database is a great, that way you have a clear target for converting to and from. So regardless of the user's preferred timezone, we should convert to UTC for storage; when retrieving the value from storage, we should convert it to the member's preferred timezone for display.

While I have covered the input from the user here, I think that I have overlooked the output...

I think that we could employ a middleware to curry the request context with the correct timezone for the user or fall back to UTC if not authenticated.

Reply:
yeah agreed! I'm still trying to get my head inside what the conversion does so I think I just misunderstood it. Middleware seems like a good approach to me :)